### PR TITLE
Refactor/kws invite user

### PR DESCRIPTION
**Pull request by GuilhermeMota93**
## JIRA
Link of the JIRA tickets the PR relates to. 
[KWS-1891](https://jira.superawesome.tv/browse/KWS-1891)
[KWS-1908](https://jira.superawesome.tv/browse/KWS-1908)

## DESCRIPTION
Added Invite user functionality as per new architecture.
Added tests for request, mapping and provider.

## SCREENSHOTS
Attach any screenshot or .gif file to show how the PR fixes an issue or the new behaviour introduced by it.
Put`n/a` if none

### CODE REVIEW

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 1]
```diff
@@ -23,50 +23,52 @@ class KWSSwiftTableViewController: UITableViewController {
...
+                ["Get App Data", "Set App Data"],
+            "Invite" :
+                ["Invite User"]
     ]
-
+    
     // MARK: - TABLE VIEW STUFF
     override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
         let textForSection = Array(functionalitiesDict)[section].key
         return "\(textForSection)"
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: change this to return textForSection without the "" and the \() it is not needed

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 2]
```diff
@@ -23,50 +23,52 @@ class KWSSwiftTableViewController: UITableViewController {
...
-
+        
         let cell = tableView.dequeueReusableCell(withIdentifier: "LabelCell", for: indexPath)
-
+        
         let listOfItemsInRow = Array(functionalitiesDict)[indexPath.section].value
         let textForRow = Array(listOfItemsInRow)[indexPath.row]
-
+        
         cell.textLabel?.text = "\(textForRow)"
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: Change this to: ```
cell.textLabel?.text = textForRow
```

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 3]
```diff
@@ -23,50 +23,52 @@ class KWSSwiftTableViewController: UITableViewController {
...
-
+        
         return cell
     }
-
+    
     override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
         return 44
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: I would add the following to the top of the class:
```
private let rowHeight: CGFloat = 44.0
```
and then return it here:
```
return rowHeight
```

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 4]
```diff
@@ -23,50 +23,52 @@ class KWSSwiftTableViewController: UITableViewController {
...
     }
-
+    
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return 50
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: I would add the following to the top of the class:
```
private let headerHeight: CGFloat = 50.0
```
and then return it here:
```
return headerHeight
```

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 5]
```diff
@@ -179,20 +187,20 @@ class KWSSwiftTableViewController: UITableViewController {
             print("No valid user cached!!!")
         }
     }
-
+    
     func updateParentEmail(){
-
+        
         let parentEmail = "guilherme.mota@superawesome.tv"
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: definitely remove this!!! I've already signed you up to dozens of dodgy mailing lists ;)

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 6]
```diff
@@ -255,19 +263,19 @@ class KWSSwiftTableViewController: UITableViewController {
...
+            
             let userId = cachedUser.id as? Int ?? 0
             let appId = cachedUser.tokenData.appId
             let token = cachedUser.token
-
+            
             userActions?.getAppData(userId: userId, appId: appId, token: token) { appData, error in
-
+                
                 if(appData != nil){
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: add space and remove brackets ;) do what ever suits you best for now! No more comments like this from now on

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 7]
```diff
@@ -278,39 +286,63 @@ class KWSSwiftTableViewController: UITableViewController {
...
             let key = "new_value"
-
+            
             userActions?.setAppData(value: value, key: key, userId: userId, appId: appId, token: token) { error in
-
+                
                 if (error == nil){
                     print("App Data set with success!")
                 } else {
                     print("Something went wrong for App Data:  \(String(describing: error))")
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: there is an extra space here: `print("Something went wrong for App Data: ^^^^^^ \(String(describing: error))")`

File: Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift [hunk 8]
```diff
@@ -278,39 +286,63 @@ class KWSSwiftTableViewController: UITableViewController {
...
+        }
+    }    
+    
     func saveUser(user: LoggedUserModelProtocol) {
         let sessionsService = KWSSDK.getService(value: SessionServiceProtocol.self, environment: kUserKWSNetworkEnvironment!)
         let success = sessionsService?.saveLoggedUser(user: user)
     }
-
+    
     func getLoggedUser () -> LoggedUser? {
```
Comment (me2k) [inline Example/KWSiOSSDKObjC/KWSSwiftTableViewController.swift]: i would remove the space between the function name and brackets, you've used that convention elsewhere

File: Example/Tests/UserActions_InviteUser+ObjectProvider.swift [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+                                                expect(error).toNot(beNil())
+                                                expect((error as! ErrorWrapper).code).to(equal(1))
+                                                expect((error as! ErrorWrapper).codeMeaning).to(equal("forbidden"))
+                                                expect((error as! ErrorWrapper).message).to(equal("operation not supported for this user"))
+                                                
+                                                done()
+            })
+        }
+    }
+
```
Comment (me2k) [inline Example/Tests/UserActions_InviteUser+ObjectProvider.swift]: extra line here, remove

File: Example/Tests/fixtures.bundle/invite_user_bad_email_response.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (me2k) [inline Example/Tests/fixtures.bundle/invite_user_bad_email_response.json]: normally JSON doesn't start with a new line, do your tests work with this file?

**Comment by GuilhermeMota93** [comment]
Will merge this in and do the View Controller changes on the next PR if that's ok @me2k.

**Pull request merged**